src/server/dflycmd.cc

```diff
@@ -63,25 +65,18 @@ struct TransactionGuard {
 };
 }  // namespace
 
-DflyCmd::ReplicaRoleInfo::ReplicaRoleInfo(std::string address, uint32_t listening_port,
-                                          SyncState sync_state, uint64_t lsn_lag)
-    : address(address), listening_port(listening_port), lsn_lag(lsn_lag) {
+std::string_view DflyCmd::SyncStateName(DflyCmd::SyncState sync_state) {
   switch (sync_state) {
-    case SyncState::PREPARATION:
-      state = "preparation";
-      break;
-    case SyncState::FULL_SYNC:
-      state = "full_sync";
-      break;
-    case SyncState::STABLE_SYNC:
-      state = "stable_sync";
-      break;
-    case SyncState::CANCELLED:
-      state = "cancelled";
-      break;
-    default:
-      break;
-  }
+    case DflyCmd::SyncState::PREPARATION:
+      return "preparation";
+    case DflyCmd::SyncState::FULL_SYNC:
+      return "full_sync";
+    case DflyCmd::SyncState::STABLE_SYNC:
+      return "stable_sync";
+    case DflyCmd::SyncState::CANCELLED:
+      return "cancelled";
+  }
+  LOG(FATAL) << "Unspported state " << int(sync_state);
```

Comment:
Better use log error here.
f.e if you add a new state and start a replica with the new version while master is in the old version were this state is not supported you will get a crash

Reply:
It's controlled by the state machine code in the same binary, it's not input from the other client.

Reply:
you are right. I would still crash in debug mode and in release just return unsupported

Reply:
ping here